tools/topic_tools/test/test_transform.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2008, Willow Garage, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of Willow Garage, Inc. nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import unittest
+import rospy
+import rostest
+import sys
+from std_msgs.msg import *
+
+
+PKG = 'topic_tools'
+NAME = 'transform_sub'
+
+
+class TransformSub(unittest.TestCase):
+
+  def msg_cb(self, msg):
+    self.msg = msg
+
+  def test_transform_sub(self):
+    rospy.init_node(NAME)
+    value = rospy.get_param("~value", 1.0)
+
+    self.msg = None
+    sub = rospy.Subscriber("input", Float32, self.msg_cb)
+
+    while not self.msg:
+      rospy.sleep(1.0)
```

Comment:
magic `1.0`  .. this might introduce random fails on any hardware, when it is busy.

I suggest to use [wait_for_message](https://docs.ros.org/melodic/api/rospy/html/rospy.client-module.html#wait_for_message) with a high timeout parameter, and let rostests's `time-limit` handle the case that the message does not arrive at all.

Reply:
`wait_for_message` makes the test actually much easier. the test is inherently a bit flaky. It seems, that sometimes rostopic does not advertise the topic early enough. Therefor the tests are configured to do three retries. The timeout in the tests should be no problem here.